hex/lib/dependabot/hex/update_checker.rb

```diff
@@ -221,13 +221,23 @@ def latest_release_from_hex_registry
               ignore_requirements.any? { |r| r.satisfied_by?(v) }
             end
 
-            raise AllVersionsIgnored if @raise_on_ignored && filtered.empty? && versions.any?
+            if @raise_on_ignored && filter_lower_versions(filtered).empty? && filter_lower_versions(versions).any?
+              raise AllVersionsIgnored
+            end
 
             filtered.max
           end
       end
       # rubocop:enable Metrics/PerceivedComplexity
 
+      def filter_lower_versions(versions_array)
+        versions_array.select do |version|
+          next unless current_version
```

Comment:
```suggestion
        return [] unless current_version

        versions_array.select do |version|
```

Nit: `current_version` isn't going to change inside the loop, so check it once and exit early.

Reply:
Nit: should we care about the memoization of `current_version`? We could shave some nanos by storing it outside the loop.